### DISCUSSION TITLE
TypeError: The "original" argument must be of type Function

### DISCUSSION THREAD

**Discussion by jsartisan** [tailwindlabs/tailwindcss#5601] [Help]
Hi there, Pawan this side. I am a frontend developer at appsmith ( https://appsmith.com/ ) and I am pushing my team to start migrating from **styled components to tailwindcss.**  We will be creating our very own UI library based on tailwindcss soon. Also, on appsmith, we have plans to introduce a feature to give our users a way to customize the appsmith widgets through tailwindcss syntax. But to do so, I need to compile tailwindcss on the browser ( because every application on appsmith can have its own theme with custom values other than default ones ).

So I was trying to compile the tailwind css via a web worker just like you do in play.tailwindcss.com but I am using create-react-app with craco ( **postcss 7 and tailwindcss/postcss7-compat** ). and  I constantly get `**The "original" argument must be of type Function**` even if I just import tailwindcss in the react code like this:

```
import "./App.css";
import tailwindcss from "tailwindcss";

function App() {
  return <div>Hi</div>;
}

export default App;
```
![image](https://user-images.githubusercontent.com/6636360/134804295-53e02682-2978-48b3-818c-80a944e35676.png)

node version: **v14.16.0**
npm version: **7.11.2**

Is there any solution to this? More importantly, will this even work? I see this works fine on https://codesandbox.io/ ( **https://codesandbox.io/s/quizzical-einstein-27bn1?file=/src/App.js** ) and play.tailwindcss.com

**Comment by adamwathan** [comment]
Hey! We only use the PostCSS 8 build ourselves when hacking things to make it work in the browser so not sure what advice I can offer here unfortunately other than to just dig in and see what you can figure out. Tailwind is a node.js build tool so while getting it running in the browser is cool and fun, I don't consider any challenges in this area to be bugs in Tailwind and they aren't things I can really justify working on unfortunately.

**Reply by jsartisan** [reply]
Okay, then it seems I'll be stuck on this till the time postcss 8 supports comes for  CRA ( https://github.com/facebook/create-react-app/issues/9664 ). Anyways, thanks @adamwathan for taking out time to reply to this query.
